### PR TITLE
Feature/nsa 6016 job banner

### PR DESCRIPTION
**Pull request by stottidfe**
https://dfe-secureaccess.atlassian.net/browse/NSA-6016 ansure that the user banner is available for the first 24 hours unless closed by user

### CODE REVIEW

File: src/app/home/getServices.js [hunk 1]
```diff
@@ -242,14 +250,15 @@ const getServices = async (req, res) => {
 
   const userPireanServices = services.filter((value) => pireanServices.includes(value.name));
   ///testing here for new services use sortby accessGranted to get the latest one
-  let nowDate = new Date().getDate();
+  let nowDate = new Date().getDate(); //used to determine last logged 24 hrs
```
Comment (malikimran34) [inline src/app/home/getServices.js:253]: its minor but add comment on top of code line. keep consistency

File: src/app/home/getServices.js [hunk 2]
```diff
@@ -242,14 +250,15 @@ const getServices = async (req, res) => {
...
   const checklastestaddition = services.filter((x) => {
     return (
-      new Date(x.accessGrantedOn).getDate() >= new Date(last_login).getDate() &&
+      new Date(x.accessGrantedOn).getDate() >= new Date(user.claims.last_login).getDate() &&
       new Date(x.accessGrantedOn).getDate() <= nowDate
     );
   });
   if (checklastestaddition.length > 0 && (res.locals.flash === undefined || res.locals.flash.title === undefined)) {
+    ///does this exist in userbanner?
```
Comment (malikimran34) [inline src/app/home/getServices.js:261]: do we need the comment line? if yes please explain in more detail

File: src/app/home/views/reviewUsers.ejs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
         <script>
             $(function() {
-                loadReviewUsersBanner();
+                //loadReviewUsersBanner();
```
Comment (malikimran34) [inline src/app/home/views/reviewUsers.ejs:9]: remove script tag if do not require this function anymore

File: src/app/requestService/approveServiceRequest.js [hunk 4]
```diff
@@ -3,7 +3,11 @@ const { listRolesOfService, addUserService, getServicesForUser } = require('../.
 const { getOrganisationAndServiceForUser } = require('../../infrastructure/organisations');
 const { getUserDetails } = require('../users/utils');
 const { actions } = require('../constans/actions');
-
+const {
+  createSubServiceAddedBanners,
```
Comment (malikimran34) [inline src/app/requestService/approveServiceRequest.js:7]: some extra imported variables

File: src/app/requestService/approveServiceRequest.js [hunk 5]
```diff
@@ -215,7 +219,7 @@ const post = async (req, res) => {
     viewModel.service.roles.map((i) => i.name),
     mngUserOrgPermission,
   );
-
+  const bannerTing = await createUserBanners(req.user.id, -3);
```
Comment (malikimran34) [inline src/app/requestService/approveServiceRequest.js:222]: do you mean bannerTag? can you please explain in comments what is -3

File: src/app/users/utils.js [hunk 6]
```diff
@@ -195,7 +195,16 @@ const isOrgEndUser = (userOrganisations, orgId) => {
   }
   return false;
 };
-
+const isLoginOver24 = (last_login, prev_login) => {
+  let a = moment(last_login, "HH:mm:ss")
+  let b = moment(prev_login, "HH:mm:ss")
+  let checkfor24 = a.diff(b, 'hours');
```
Comment (malikimran34) [inline src/app/users/utils.js:201]: check in minutes to be precise

**Review by malikimran34 — CHANGES_REQUESTED** [review]

File: src/app/home/getServices.js (hunk 2 above)
Comment (stottidfe) [reply src/app/home/getServices.js:261]: its a place holder for the next job I take whoich is to do with displaying the banner for new services for only twentyfour hours and persistent close option for user

File: src/app/home/views/reviewUsers.ejs (hunk 3 above)
Comment (stottidfe) [reply src/app/home/views/reviewUsers.ejs:9]: this is not my work bud and was stopping me working so added comment do you want me to remove the commented out line

File: src/app/requestService/approveServiceRequest.js (hunk 4 above)
Comment (stottidfe) [reply src/app/requestService/approveServiceRequest.js:7]: not my work pulled in form develop?

File: src/app/home/views/reviewUsers.ejs (hunk 3 above)
Comment (stottidfe) [reply src/app/home/views/reviewUsers.ejs]: added back after testing

File: src/app/requestService/approveServiceRequest.js (hunk 5 above)
Comment (stottidfe) [reply src/app/requestService/approveServiceRequest.js]: removed

**Review by malikimran34 — APPROVED** [review]

**Pull request merged**